sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/VectorIndexSpec.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Vector Indexes spec for Azure CosmosDB service.
+ */
+public final class VectorIndexSpec {
+
+    private final JsonSerializable jsonSerializable;
+    private String type;
+
+    /**
+     * Constructor
+     *
+     * @param path the path.
+     */
+    public VectorIndexSpec(String path) {
+        this.jsonSerializable = new JsonSerializable();
+        this.setPath(path);
+    }
+
+    /**
+     * Constructor.
+     */
+    public VectorIndexSpec() {
+        this.jsonSerializable = new JsonSerializable();
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param objectNode the object node that represents the included path.
+     */
+    public VectorIndexSpec(ObjectNode objectNode) { this.jsonSerializable = new JsonSerializable(objectNode); }
+
+    /**
+     * Gets path.
+     *
+     * @return the path.
+     */
+    public String getPath() {
```

Comment:
we can hide public API if not required.

Reply:
getter and setter should be available both for cnsistency - so, even if getPath isn't used I would keep it here.

Reply:
Retaining the getters as they are is essential for consistency, as per Fabian's suggestion.